### PR TITLE
fix: make sure we scale image before trying to compress it

### PR DESCRIPTION
**Pull request by Bilb**
looks like the scaling of images was broken and so we were compressing the original picture, to the desired size (kb) but keeping it's original width and height. Which meant we were compressing them way too much
fixing this:
image
to be this:
image

### CODE REVIEW

File: ts/util/attachmentsUtil.ts [hunk 1]
```diff
@@ -189,44 +189,52 @@ export async function autoScale<T extends { contentType: string; blob: Blob }>(
...
-    crop: !!makeSquare,
-    orientation: 1,
-    aspectRatio: makeSquare ? 1 : undefined,
-    canvas: true,
-    imageSmoothingQuality: 'medium',
-  };
-
   perfStart(`loadimage-*${blob.size}`);
-  const canvas = await loadImage(blob, loadImgOpts);
+  const canvasPlop = await loadImage(blob, {});
```
Comment (yougotwill) [inline ts/util/attachmentsUtil.ts:193]: Needs updating

File: ts/util/attachmentsUtil.ts [hunk 2]
```diff
@@ -189,44 +189,52 @@ export async function autoScale<T extends { contentType: string; blob: Blob }>(
...
+  const canvasOri = loadImage.scale(
```
Comment (yougotwill) [inline ts/util/attachmentsUtil.ts:194]: Origin or Original?

File: ts/util/attachmentsUtil.ts [hunk 3]
```diff
@@ -189,44 +189,52 @@ export async function autoScale<T extends { contentType: string; blob: Blob }>(
...
+      crop: !!makeSquare,
+      cover: !!makeSquare,
+      orientation: 1,
+      // aspectRatio: makeSquare ? 1 : undefined,
+      canvas: true,
+      imageSmoothingQuality: 'medium',
+      meta: false,
+    }
+  );
+  if (maxMeasurements?.maxSide) {
```
Comment (yougotwill) [inline ts/util/attachmentsUtil.ts:208]: This needs cleanup or a comment to explain when we need this for debugging purposes

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/util/attachmentsUtil.ts (hunk 3 above)
Comment (Bilb) [reply ts/util/attachmentsUtil.ts:208]: removed

**Review by yougotwill — APPROVED** [review]
LGTM

**Comment by Bilb** [comment]
just tagging @burtonemily for tracking

**Comment by burtonemily** [comment]
Found some issues https://optf.atlassian.net/browse/SES-1244

File: ts/util/attachmentsUtil.ts [hunk 4]
```diff
@@ -189,44 +189,59 @@ export async function autoScale<T extends { contentType: string; blob: Blob }>(
...
+  const canvasLoad = await loadImage(blob, {});
+  const canvasScaled = loadImage.scale(
+    canvasLoad.image, // img or canvas element
+    {
+      maxWidth: makeSquare ? maxMeasurements?.maxSide : maxWidth,
+      maxHeight: makeSquare ? maxMeasurements?.maxSide : maxHeight,
+      crop: !!makeSquare,
+      cover: !!makeSquare,
+      orientation: 1,
+      // aspectRatio: makeSquare ? 1 : undefined,
```
Comment (yougotwill) [inline ts/util/attachmentsUtil.ts:202]: leaving this as reference?

File: ts/util/attachmentsUtil.ts [hunk 5]
```diff
@@ -240,10 +255,10 @@ export async function autoScale<T extends { contentType: string; blob: Blob }>(
   do {
     i -= 1;
     if (DEBUG_ATTACHMENTS_SCALE) {
-      // window.log.info(`autoscale iteration: [${i}] for:`, attachment);
+      window.log.info(`autoscale iteration: [${i}] for:`, JSON.stringify(readAndResizedBlob.size));
```
Comment (yougotwill) [inline ts/util/attachmentsUtil.ts:258]: Should this not be `window.log.debug` ?

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/util/attachmentsUtil.ts (hunk 5 above)
Comment (Bilb) [reply ts/util/attachmentsUtil.ts:258]: updated all of them but I still want it not enabled whenever we turn the debug flag on so I kept the boolean too

File: ts/util/attachmentsUtil.ts (hunk 4 above)
Comment (Bilb) [reply ts/util/attachmentsUtil.ts:202]: removed

**Review by yougotwill — APPROVED** [review]
LGTM! 🚢

File: ts/components/conversation/message/message-content/MessageContent.tsx [hunk 6]
```diff
@@ -154,7 +154,8 @@ export const MessageContent = (props: Props) => {
 
   const toolTipTitle = moment(serverTimestamp || timestamp).format('llll');
 
-  const isDetailViewAndSupportsAttachmentCarousel = isDetailView && canDisplayImage(attachments);
+  const isDetailViewAndSupportsAttachmentCarousel =
+    isDetailView && canDisplayImagePreview(attachments);
```
Comment (yougotwill) [inline ts/components/conversation/message/message-content/MessageContent.tsx:158]: why is this on separate lines now? 🤔

**Review by yougotwill — CHANGES_REQUESTED** [review]

**Pull request merged**